### DISCUSSION TITLE
[Item] a link to item PDF，一个方便跳转到文献页面的链接按钮

### DISCUSSION THREAD

**Discussion by nilleo** [windingwind/zotero-better-notes#588] [Note Templates]
### **If the template doesn't work, please use the code provided by the helpful friend in the comments.**
Type: Item
Description: a link to item PDF，我用来放标题旁边方便看笔记时打开对应的文献PDF。The PDF link will be displayed when the cursor is placed to the left of 🔗.

Screenshots: 
![image](https://github.com/windingwind/zotero-better-notes/assets/97679720/151626ac-8feb-4d09-a684-3f14309ce438)

To import this template:
1. Copy the template below.
<details style="text-indent: 2em">
<summary>Click to show the example template</summary>

```yaml
name: "[item] a link to item"
content: |-
  // @author nilleo
  // @link https://github.com/windingwind/zotero-better-notes/discussions/588
     <a href="${{
    async function getPDFLink(item) {
      const att = await item.getBestAttachment();
      if (!att || !att.isPDFAttachment()) {
        return "";
      }
      key = att.key;
      if (att.libraryID === 1) {
        return `zotero://open-pdf/library/items/${key}`;
      } else {
        groupID = Zotero.Libraries.get(att.libraryID).id;
        return `zotero://open-pdf/groups/${groupID}/items/${key}`;
      }
    }
    sharedObj.getPDFLink = getPDFLink;
    return await getPDFLink(topItem);
  }}$">
  🔗   <!-- the link text 链接的文字-->
      </a>
```

</details>

2. Goto Zotero menubar, click `Edit`->`New Template from Clipboard`.
3. Click OK.

**Comment by WanzerJ** [comment]
不好意思，请问为什么我这边一直显示：
>SyntaxError: missing : after property id 

我也尝试过加`:`，但没有任何作用

**Reply by Kingw413** [reply]
change it to:
```
<a href=zotero://open-pdf/0_${Zotero.Items.get(topItem.getAttachments()).filter((i) => i.isPDFAttachment())[0].key}>
              ${Zotero.Items.get(topItem.getAttachments()).filter((i)=> i.isPDFAttachment())[0].getFilename()} 
</a>
```

**Reply by WanzerJ** [reply]
The code you provided is working, thank you!

**Reply by Kingw413** [reply]
Glad to be of help, you're welcome.

**Reply by huayu123-hope** [reply]
sorry i am a fresher, can you tell me what "it" is? if you are available, can you type the complete code

**Reply by huayu123-hope** [reply]
哥们，咋改啊？我没看懂是咋改的。我试了好多次了，都不行啊

**Reply by WanzerJ** [reply]
Zotero主页面-->编辑-->模板编辑器-->新建-->名称：`[item] XXX`-->输入框内复制粘贴以下代码
```
<a href=zotero://open-pdf/0_${Zotero.Items.get(topItem.getAttachments()).filter((i) => i.isPDFAttachment())[0].key}>
              ${Zotero.Items.get(topItem.getAttachments()).filter((i)=> i.isPDFAttachment())[0].getFilename()} 
</a>
```
名称里面前缀`[item]`是固定的，好像还有其他的前缀，可以看一下[Write Note Template](https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md#write-note-template)

**Reply by huayu123-hope** [reply]
可以了，老哥。我真傻！害

**Reply by huayu123-hope** [reply]
感谢老哥

**Comment by macrospecter** [comment]
I got the same error as listed above.

**Reply by Kingw413** [reply]
change it to:
```
<a href=zotero://open-pdf/0_${Zotero.Items.get(topItem.getAttachments()).filter((i) => i.isPDFAttachment())[0].key}>
              ${Zotero.Items.get(topItem.getAttachments()).filter((i)=> i.isPDFAttachment())[0].getFilename()} 
</a>
```
and you can create it as a new template or just add this code to an existing template in anywhere.

**Comment by Kingw413** [comment]
Got the same error as listed above.
You only need to change it like this:

```HTML
<a href=zotero://open-pdf/0_${Zotero.Items.get(topItem.getAttachments()).filter((i) => i.isPDFAttachment())[0].key}>
              ${Zotero.Items.get(topItem.getAttachments()).filter((i)=> i.isPDFAttachment())[0].getFilename()} 
</a>
```

**Reply by Fontanapink** [reply]
How can I write this so that it replaces the title text with the link (paperclip) icon instead?

**Reply by Kingw413** [reply]
just change it to:
```
<a href=zotero://open-pdf/0_${Zotero.Items.get(topItem.getAttachments()).filter((i) => i.isPDFAttachment())[0].key}>
            🔗
</a>
```

**Reply by huayu123-hope** [reply]
sorry i am a fresher, can you tell me what "it" is? if you are available, can you type the complete code

**Comment by cloudwalke** [comment]
Would it be possible to have this link point to the notebook itself, rather than the PDF?

**Reply by Calorion** [reply]
> Would it be possible to have this link point to the notebook itself, rather than the PDF?

Agreed. I would like the link to be to the enclosing item, not the PDF. Is this possible?
